python_modules/dagster/dagster/_grpc/impl.py

```diff
@@ -535,42 +535,28 @@ def get_partition_set_execution_param_data(
     ) = _get_job_partitions_and_config_for_partition_set_name(repo_def, partition_set_name)
 
     try:
-        with _instance_from_ref_for_dynamic_partitions(instance_ref, partitions_def) as instance:
-            with user_code_error_boundary(
-                PartitionExecutionError,
-                lambda: (
-                    "Error occurred during the partition generation for partitioned config on job"
-                    f" '{job_def.name}'"
-                ),
-            ):
-                all_partition_keys = partitions_def.get_partition_keys(
-                    dynamic_partitions_store=instance
-                )
-                partition_keys = [key for key in all_partition_keys if key in partition_names]
```

Comment:
if we think this validation is important, using has_partition_key in a loop would be another option - O(partition_names) point checks is likely faster than enumerating every partition key

Reply:
although those checks can still involve a data fetch in the dynamic partitions case, so my bias would still be to remove it altogether